cli/main.rs

```diff
@@ -17,65 +14,87 @@ enum Mode {
     #[structopt(name = "encode")]
     Encode {
         /// The base to use for encoding.
-        #[structopt(short = "b", long = "base", default_value = "base58-btc")]
+        #[structopt(short = "b", long = "base", default_value = "base58btc")]
         base: StrBase,
+        /// The data need to be encoded.
+        #[structopt(short = "i", long = "input")]
+        input: String,
     },
     #[structopt(name = "decode")]
-    Decode,
+    Decode {
+        /// The data need to be decoded.
+        #[structopt(short = "i", long = "input")]
+        input: String,
+    },
 }
 
-fn main() -> Result<(), ExitFailure> {
-    env_logger::init();
+fn main() -> Result<()> {
     let opts = Opts::from_args();
     match opts.mode {
-        Mode::Encode { base } => encode(base),
-        Mode::Decode => decode(),
+        Mode::Encode { base, input } => encode(base, input.as_bytes()),
+        Mode::Decode { input } => decode(&input),
     }
 }
 
 #[derive(Debug)]
 struct StrBase(Base);
 
-impl fmt::Display for StrBase {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+impl std::fmt::Display for StrBase {
```

Comment:
Please try to keep changes that are not related to the actual work to a minimum. In this case, please keep the original imports.